lib/utils/apiUtils.js

```diff
@@ -59,104 +65,219 @@ async function refreshTokens(firmId) {
+      description ? `\nError description: ${description}` : "",
       "\n",
       `Error refreshing the tokens. Try running the authentication process again`
     );
     process.exit(1);
   }
 }
 
-function setAxiosDefaults(firmId) {
-  const firmTokens = firmCredentials.getTokenPair(firmId);
-  if (!firmTokens) {
-    consola.error(`Missing authorization for firm id: ${firmId}`);
-    process.exit(1);
+function setAxiosDefaults(type, envId) {
+  let axiosInstance;
+
+  if (type == "firm") {
+    const firmTokens = firmCredentials.getTokenPair(envId);
+    if (!firmTokens) {
+      consola.error(`Missing authorization for firm id: ${envId}`);
+      process.exit(1);
+    }
+
+    axiosInstance = axios.create({
+      baseURL: `${BASE_URL}/api/v4/f/${envId}`,
+      headers: {
+        "User-Agent": `silverfin-cli/${pkg.version}`,
+        Authorization: `Bearer ${firmTokens.accessToken}`,
+      },
+    });
+
+    axiosInstance.interceptors.response.use(
+      (res) => {
+        return res;
+      },
+      async (error) => {
+        const originalConfig = error.config;
+        if (error.response) {
+          if (error.response.status === 401 && !originalConfig._retry) {
+            // Set _retry to true after trying the first refresh request on 401 status to avoid an infinite loop
+            originalConfig._retry = true;
+
+            try {
+              // Get a refreshed set of tokens
+              const firmId = originalConfig.baseURL.split("/").pop();
+              await refreshTokens(axiosInstance, firmId);
+
+              // Set the Authorization header with the new access token
+              const firmTokens = firmCredentials.getTokenPair(firmId);
+              consola.debug(`refreshed tokens for firm ${firmId}`);
+
+              axiosInstance.defaults.headers.common[
+                "Authorization"
+              ] = `Bearer ${firmTokens.accessToken}`;
+
+              originalConfig.headers.Authorization = `Bearer ${firmTokens.accessToken}`;
+
+              return axiosInstance(originalConfig);
+            } catch (_error) {
+              consola.error(
+                `Error 401: Failed to refresh the firm access token automatically, try to manually authorize the firm again with the authorize command`
+              );
+
+              if (_error.response && _error.response.data) {
+                return Promise.reject(_error.response.data);
+              }
+
+              return Promise.reject(_error);
+            }
+          }
+        }
+
+        return Promise.reject(error);
+      }
+    );
+  } else if (type == "partner") {
+    axiosInstance = axios.create({
+      baseURL: `${BASE_URL}/api/partner/v1`,
+      headers: {
+        "User-Agent": `silverfin-cli/${pkg.version}`,
+      },
+    });
+
+    axiosInstance.interceptors.request.use((config) => {
+      // Fetch the stored partner api key
+      const partnerToken = firmCredentials.getPartnerCredentials(envId)?.token;
```

Comment:
`firmCredentials.getPartnerCredentials` reads weird. Not sure if refactoring is within the scope of this PR though

Reply:
This would indeed be valid for a refactor (maybe by rename of the class or moving the partnerCredentials to its separate class, although the partnerCredentials functions use some shared functions with the firmCredentials, so would need a think on what's the best). 

Not sure if we should still scope into this PR since it's already quite big & the authorization is already well-tested now and I might break smth again with a refactor 😬 

`firmCredentials` is used a lot in the VS Code extension as well, might be a good idea to refactor as part of updating the CLI version in the extension.